### ISSUE TITLE
🐛[BUG]: Resdiff conditional input scaling

### ISSUE THREAD

**Issue by nbren12** [NVIDIA/modulus#229]
### Version

n/a

### Describe the issue

I know resdiff isn't in the code base yet, but wanted to file this since I'll be gone.

Not a bug per-se, but @MortezaMardani, Arash, Jaideep and I have discussed an issue with how the conditional inputs are scaled in the EDMPreconditioner:
```
        #SR: concatenate input channels
        if img_lr is None:
            x = x
        else:
            x = torch.cat((x, img_lr), dim=1)

        x = x.to(torch.float32)
        sigma = sigma.to(torch.float32).reshape(-1, 1, 1, 1)
        class_labels = None if self.label_dim == 0 else torch.zeros([1, self.label_dim], device=x.device) if class_labels is None else class_labels.to(torch.float32).reshape(-1, self.label_dim)
        dtype = torch.float16 if (self.use_fp16 and not force_fp32 and x.device.type == 'cuda') else torch.float32

        c_skip = self.sigma_data ** 2 / (sigma ** 2 + self.sigma_data ** 2)
        c_skip = 0.0*c_skip
        c_out = sigma * self.sigma_data / (sigma ** 2 + self.sigma_data ** 2).sqrt()
        c_out = torch.ones_like(c_out)
        c_in = 1 / (self.sigma_data ** 2 + sigma ** 2).sqrt()
        c_in = torch.ones_like(c_in)
        c_noise = sigma.log() / 4
        c_noise = 0.0*c_noise
        
        F_x = self.model((c_in * x).to(dtype), c_noise.flatten(), class_labels=class_labels, **model_kwargs)
```

Specifically, for large noise c_in is very small. So both the conditional input "img_lr" and the latents are scaled down. I think we agreed the conditional inputs should probably not scaled down like this (i.e. the class labels are not).

The code would be modified like this:
```
        x = x.to(torch.float32)
        sigma = sigma.to(torch.float32).reshape(-1, 1, 1, 1)
        class_labels = (
            None
            if self.label_dim == 0
            else torch.zeros([1, self.label_dim], device=x.device)
            if class_labels is None
            else class_labels.to(torch.float32).reshape(-1, self.label_dim)
        )
        dtype = (
            torch.float16
            if (self.use_fp16 and not force_fp32 and x.device.type == "cuda")
            else torch.float32
        )

        c_skip = self.sigma_data**2 / (sigma**2 + self.sigma_data**2)
        c_out = sigma * self.sigma_data / (sigma**2 + self.sigma_data**2).sqrt()
        c_in = 1 / (self.sigma_data**2 + sigma**2).sqrt()
        c_noise = sigma.log() / 4

        arg = c_in * x

        if img_lr is not None:
            arg = torch.cat([arg, img_lr], dim=1)

        F_x = self.model(
            arg.to(dtype),
            c_noise.flatten(),
            class_labels=class_labels,
            **model_kwargs,
        )
```

cc @mnabian

**Comment by jleinonen** [comment]
Would this change break backward compatibility with checkpoints created with earlier versions?

**Comment by nbren12** [comment]
I think #373  is incorrect.
